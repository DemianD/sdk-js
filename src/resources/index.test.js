import * as resources from '.'

describe('resources', () => {
  it('should define all the resources', () => {
    expect(resources).toMatchSnapshot()
  })
})
