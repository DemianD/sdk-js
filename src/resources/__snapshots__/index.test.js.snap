// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resources should define all the resources 1`] = `
Object {
  "accessTokens": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "access_tokens",
    "sub_resources": Object {},
  },
  "admins": Object {
    "allowed_includes": Array [
      "user",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "admins",
    "sub_resources": Object {},
  },
  "appointments": Object {
    "allowed_includes": Array [
      "subject",
      "subject.subject_category",
      "office",
      "customer",
      "contact",
      "contact.employee",
      "contact.user",
      "possibilities",
      "possibilities.created_by",
      "accepted_possibility",
      "answers",
      "initiated_by",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "state",
      "start",
      "end",
    ],
    "headers": Object {
      "post": Object {
        "recaptcha": [Function],
      },
    },
    "method": undefined,
    "resource": "appointments",
    "sub_resources": Object {
      "answers": Object {
        "allowed_includes": Array [
          "question",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "answers",
        "sub_resources": Object {},
      },
      "availableContacts": Object {
        "allowed_includes": Array [
          "user",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "available_contacts",
        "sub_resources": Object {},
      },
      "messages": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "messages",
        "sub_resources": Object {},
      },
      "possibilities": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "possibilities",
        "sub_resources": Object {},
      },
    },
  },
  "centralPlanners": Object {
    "allowed_includes": Array [
      "user",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "id",
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "central_planners",
    "sub_resources": Object {},
  },
  "contacts": Object {
    "allowed_includes": Array [
      "user",
      "offices",
      "offices.subject_settings",
      "subjects",
      "subjects.subject_category",
      "appointments",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "offices",
      "schedulable",
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "contacts",
    "sub_resources": Object {
      "appliedDayTemplates": Object {
        "allowed_includes": Array [
          "day_templates",
          "day_templates.day_time_slots",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [
          "start",
          "end",
        ],
        "headers": Object {},
        "method": undefined,
        "resource": "applied_day_templates",
        "sub_resources": Object {},
      },
      "appliedWeekTemplates": Object {
        "allowed_includes": Array [
          "week_templates",
          "week_templates.week_time_slots",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [
          "start",
          "end",
          "split",
        ],
        "headers": Object {},
        "method": undefined,
        "resource": "applied_week_templates",
        "sub_resources": Object {},
      },
      "appointments": Object {
        "allowed_includes": Array [
          "possibilities",
          "accepted_possibility",
          "customer",
          "subject",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "appointments",
        "sub_resources": Object {},
      },
      "availability": Object {
        "allowed_includes": Array [
          "offices",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [
          "office",
          "start",
          "end",
        ],
        "headers": Object {},
        "method": undefined,
        "resource": "availability",
        "sub_resources": Object {},
      },
      "dayTemplates": Object {
        "allowed_includes": Array [
          "day_time_slots",
          "day_time_slots.offices",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "day_templates",
        "sub_resources": Object {
          "dayTimeSlots": Object {
            "allowed_includes": Array [
              "offices",
            ],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "day_time_slots",
            "sub_resources": Object {},
          },
        },
      },
      "offices": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "offices",
        "sub_resources": Object {
          "availabilityHours": Object {
            "allowed_includes": Array [],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "availability_hours",
            "sub_resources": Object {},
          },
          "availabilityHoursExceptions": Object {
            "allowed_includes": Array [],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "availability_hours_exceptions",
            "sub_resources": Object {},
          },
        },
      },
      "subjects": Object {
        "allowed_includes": Array [
          "questions",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "subjects",
        "sub_resources": Object {},
      },
      "weekTemplates": Object {
        "allowed_includes": Array [
          "week_time_slots",
          "week_time_slots.offices",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "week_templates",
        "sub_resources": Object {
          "weekTimeSlots": Object {
            "allowed_includes": Array [
              "offices",
            ],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "week_time_slots",
            "sub_resources": Object {},
          },
        },
      },
    },
  },
  "customers": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "external_id",
      "match",
      "limit",
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "customers",
    "sub_resources": Object {},
  },
  "employees": Object {
    "allowed_includes": Array [
      "user",
      "enterprise",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "email",
      "external_id",
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "employees",
    "sub_resources": Object {},
  },
  "enterpriseSettings": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "enterprise_settings",
    "sub_resources": Object {},
  },
  "enterprises": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "enterprises",
    "sub_resources": Object {},
  },
  "identity": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "get",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "identity",
    "sub_resources": Object {},
  },
  "insights": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "insights",
    "sub_resources": Object {
      "auth": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "post",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "auth",
        "sub_resources": Object {},
      },
    },
  },
  "integrations": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      Object {
        "key": "filter[type]",
        "name": "type",
      },
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "integrations",
    "sub_resources": Object {},
  },
  "languages": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "languages",
    "sub_resources": Object {},
  },
  "modules": Object {
    "allowed_includes": Array [],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "modules",
    "sub_resources": Object {
      "languages": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "languages",
        "sub_resources": Object {
          "translations": Object {
            "allowed_includes": Array [],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "translations",
            "sub_resources": Object {},
          },
        },
      },
      "translations": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "translations",
        "sub_resources": Object {},
      },
    },
  },
  "officeManagers": Object {
    "allowed_includes": Array [
      "user",
      "offices",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "office_managers",
    "sub_resources": Object {
      "offices": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "offices",
        "sub_resources": Object {},
      },
    },
  },
  "offices": Object {
    "allowed_includes": Array [
      "subject_settings",
      "contacts",
      "contacts.employee",
      "contacts.subjects",
      "contacts.user",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "external_id",
      "schedulable_for_subject",
      "schedulable_with_contact",
      "schedulable",
      "nearest_to_geo",
      "limit",
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "offices",
    "sub_resources": Object {
      "availabilityHoursExceptions": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "availability_hours_exceptions",
        "sub_resources": Object {},
      },
      "contacts": Object {
        "allowed_includes": Array [
          "user",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "contacts",
        "sub_resources": Object {},
      },
      "officeManagers": Object {
        "allowed_includes": Array [
          "user",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "office_managers",
        "sub_resources": Object {},
      },
      "subjectSettings": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [
          "subject_id",
        ],
        "headers": Object {},
        "method": undefined,
        "resource": "subject_settings",
        "sub_resources": Object {},
      },
    },
  },
  "subjectCategories": Object {
    "allowed_includes": Array [
      "languages",
      "languages.translations",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "subject_categories",
    "sub_resources": Object {},
  },
  "subjects": Object {
    "allowed_includes": Array [
      "questions",
      "subject_category",
      "languages",
      "languages.translations",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [
      "category_id",
      "external_id",
      "schedulable_at_office",
      "schedulable_with_contact",
      "schedulable",
    ],
    "headers": Object {},
    "method": undefined,
    "resource": "subjects",
    "sub_resources": Object {
      "questions": Object {
        "allowed_includes": Array [
          "options",
          "languages",
          "languages.translations",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "questions",
        "sub_resources": Object {
          "languages": Object {
            "allowed_includes": Array [],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "languages",
            "sub_resources": Object {},
          },
          "options": Object {
            "allowed_includes": Array [],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [],
            "headers": Object {},
            "method": undefined,
            "resource": "options",
            "sub_resources": Object {},
          },
        },
      },
      "timetable": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [
          "office",
          "start",
          "end",
          "contacts",
        ],
        "headers": Object {},
        "method": undefined,
        "resource": "timetable",
        "sub_resources": Object {},
      },
    },
  },
  "users": Object {
    "allowed_includes": Array [
      "employees",
      "sync_accounts",
    ],
    "allowed_methods": Array [
      "*",
    ],
    "filters": Array [],
    "headers": Object {},
    "method": undefined,
    "resource": "users",
    "sub_resources": Object {
      "calendars": Object {
        "allowed_includes": Array [
          "sync_accounts",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "calendars",
        "sub_resources": Object {
          "events": Object {
            "allowed_includes": Array [],
            "allowed_methods": Array [
              "*",
            ],
            "filters": Array [
              "start",
              "end",
              "split",
              "group_by",
            ],
            "headers": Object {},
            "method": undefined,
            "resource": "events",
            "sub_resources": Object {},
          },
        },
      },
      "employees": Object {
        "allowed_includes": Array [
          "roles",
        ],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "employees",
        "sub_resources": Object {},
      },
      "events": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [
          "start",
          "end",
          "split",
          "group_by",
        ],
        "headers": Object {},
        "method": undefined,
        "resource": "events",
        "sub_resources": Object {},
      },
      "syncAccounts": Object {
        "allowed_includes": Array [],
        "allowed_methods": Array [
          "*",
        ],
        "filters": Array [],
        "headers": Object {},
        "method": undefined,
        "resource": "sync_accounts",
        "sub_resources": Object {},
      },
    },
  },
}
`;
